src/utils/InternetAddress.php

```diff
@@ -17,27 +17,33 @@
 
 namespace raklib\utils;
 
-class InternetAddress{
+use function inet_pton;
+use function strlen;
 
-	/**
-	 * @var string
-	 */
+class InternetAddress{
+	/** @var string */
 	public $ip;
-	/**
-	 * @var int
-	 */
+	/** @var int */
 	public $port;
-	/**
-	 * @var int
-	 */
+	/** @var int */
 	public $version;
 
 	public function __construct(string $address, int $port, int $version){
+		$encodedAddress = inet_pton($address);
+		if($encodedAddress === false){
+			throw new \InvalidArgumentException("Failed to parse internet address");
+		}
```

Comment:
Is it really necessary to throw exception here instead of in the function that actually encodes the address?

Reply:
It makes sense to validate user input. User may pass wrong port/address here:
https://github.com/pmmp/PocketMine-MP/blob/f5bf93455a87f78da31c68d5915adb0054b91b68/src/Server.php#L361
Then it throws an exception and and will not let start the server
https://github.com/pmmp/PocketMine-MP/blob/f5bf93455a87f78da31c68d5915adb0054b91b68/src/network/mcpe/raklib/RakLibInterface.php#L97

Reply:
That sounds like the responsibility of the config provider to validate user input. Parsing the address is not a cost-free operation, and there is no need to parse the check twice if you are going to call putIp() on every InternetAddress instance anyway.

Reply:
Half the point of having a type that encapsulates this information is so that you can be sure you're getting valid input when typehinting `InternetAddress`. That way you don't need to litter the code with `isValid()` checks.

Reply:
Is InternetAddress used for anything other than encoding? Could we encode it directly in the constructor? I don't like the fact that we have to parse it twice just for checking and encoding, with probably different logic.

Reply:
Yes, it's used in several places.